src/Map.ts

```diff
@@ -111,7 +111,7 @@ export namespace Shim {
 		 * @param key The key to look up
 		 * @return The value if one exists or undefined
 		 */
-		get(key: K): V {
+		get(key: K): V | undefined {
```

Comment:
With the shim's we should also match the exported default class on these changes.

We should also reflect this in the `src/native/Map` as well.

Also, you might want to validate everything against the [lib.es6.d.ts](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.es6.d.ts#L4619)